quickshop-bukkit/src/main/java/com/ghostchu/quickshop/menu/ShopBrowseMenu.java

```diff
@@ -18,32 +18,52 @@
  */
 
 import com.ghostchu.quickshop.QuickShop;
-import com.ghostchu.quickshop.menu.browse.MainPage;
+import com.ghostchu.quickshop.menu.browse.GroupedItemPage;
+import com.ghostchu.quickshop.menu.browse.ShopListPage;
+import com.ghostchu.quickshop.menu.config.GuiConfig;
 import net.tnemc.menu.core.Menu;
 import net.tnemc.menu.core.Page;
-import net.tnemc.menu.core.PlayerInstancePage;
 
 /**
- * ShopBrowseMenu
+ * ShopBrowseMenu - Enhanced market browser with grouping, filtering, sorting, and search
  *
  * @author creatorfromhell
  * @since 6.2.0.8
  */
 public class ShopBrowseMenu extends Menu {
 
+  // Data keys for viewer state
   public static final String SHOPS_DATA = "BROWSE_SHOPS";
   public static final String SHOPS_PAGE = "BROWSE_PAGE";
+  public static final String BROWSE_SORT = "BROWSE_SORT";
+  public static final String BROWSE_FILTER = "BROWSE_FILTER";
+  public static final String BROWSE_SEARCH = "BROWSE_SEARCH";
+  public static final String BROWSE_STOCK_ONLY = "BROWSE_STOCK_ONLY";
+  public static final String BROWSE_WORLD_ONLY = "BROWSE_WORLD_ONLY";
+  
+  // Data keys for shop list page (when viewing shops for a specific item)
+  public static final String SELECTED_ITEM_SHOPS = "SELECTED_ITEM_SHOPS";
+  public static final String SHOP_LIST_PAGE = "SHOP_LIST_PAGE";
 
   public ShopBrowseMenu() {
 
-    this.rows = 6;
+    // Load rows from GUI config
+    final GuiConfig.MenuConfig menuConfig = QuickShop.getInstance().getGuiConfig().getMenuConfig("browse");
+    this.rows = menuConfig != null ? menuConfig.getRows() : 6;
```

Comment:
The ternary condition should have parentheses around it, with the question mark operator having no space before it.